tools/pipeline_perf_test/load_generator/loadgen.py

```diff
@@ -209,18 +222,144 @@ def worker_thread(self, thread_id: int, args: dict) -> None:
+            syslog_message = self.create_syslog_message(
+                body_size=args["body_size"],
+                num_attributes=args["num_attributes"],
+                attribute_value_size=args["attribute_value_size"],
+            )
+            syslog_batch.append(syslog_message)
+
+        # Combine all messages into a single buffer for efficient sending
+        batch_buffer = b''.join(syslog_batch)
+        batch_total_size = len(batch_buffer)
+
+        next_send_time = time.perf_counter()
+        while not self.stop_event.is_set():
+            try:
+                sock.sendall(batch_buffer)
+                self.increment_metric("sent", args["batch_size"])
+                self.increment_metric("bytes_sent", batch_total_size)
+            except Exception as e:
+                print(f"Thread {thread_id}: Failed to send syslog batch: {e}")
+                self.increment_metric("failed", args["batch_size"])
+                # Try to reconnect
+                try:
+                    sock.close()
+                    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+                    sock.settimeout(5)
+                    sock.connect((syslog_server, syslog_port))
+                except Exception as reconnect_error:
+                    print(f"Thread {thread_id}: Reconnection failed: {reconnect_error}")
+                    break
+
+            # If we're targeting a specific rate we do additional calculations
+            # to ensure we're not exceeding it via sleep. If we're not reaching
+            # the target rate (e.g. we're sending without sleep and it's
+            # still too slow), we increment a metric to inform observers.
+            if batch_interval:
+                now = time.perf_counter()
+                sleep_time = next_send_time - now
+                if sleep_time > 0:
+                    time.sleep(sleep_time)
+                elif now - next_send_time > batch_interval:
+                    # More than 1 interval behind
+                    self.increment_metric("late_batches")
+                next_send_time += batch_interval
+
+        sock.close()
+
+    def create_syslog_message(
+        self,
+        body_size: int = 25,
+        num_attributes: int = 2,
+        attribute_value_size: int = 15,
+    ) -> bytes:
+        """
+        Create a single syslog message with structure similar to OTLP log record.
+        """
+        hostname = socket.gethostname()
+
+        # Pre-generate static parts of the message
+        pri = "<134>"  # local0.info = 16*8+6 = 134
+        tag = "loadgen"
+
+        # Generate timestamp (RFC3164 format with space-padded day)
+        utc_time = dt.now(timezone.utc)
+        day = utc_time.day
+        timestamp = utc_time.strftime(f"%b {day:2d} %H:%M:%S")
+
+        # Create log message body (similar to OTLP body)
+        log_message = self.generate_random_string(body_size)
+
+        # Create attributes (similar to OTLP attributes)
```

Comment:
This is not needed. There is not attribute parsing done within the message contents.

Reply:
removed.